marklogic-data-hub-central/ui/src/components/results-tabular-view/results-tabular-view.tsx

```diff
@@ -154,7 +154,7 @@ const ResultsTabularView = (props) => {
         let primaryKeyValue = item.primaryKey?.propertyValue;
         let isUri = item.primaryKey?.propertyPath === 'uri';
         let uri = encodeURIComponent(item.uri);
-        let path = { pathname: `/detail/${isUri ? '-' : primaryKeyValue}/${uri}` };
+        let path = { pathname: `/detail/${isUri ? '-' : encodeURI(primaryKeyValue)}/${uri}` };
```

Comment:
Why are we using primaryKeyValue within the function encodeURI ? If 'encodeURI' is a generic function to encode any property then should we rename it to something like 'encodeProperty' ?

Reply:
encodeURI is a standard method to code a string for URL use. But since it's a string and not the full URL, the method should be encodeURIComponent.

Reply:
It actually should be encodeURIComponent not encodeURI. Thanks for noticing that. It's JS global function.